src/metabase/db/data_migrations.clj

```diff
@@ -334,6 +336,64 @@
       (db/update-where! model {:collection_id nil}
         :collection_id (u/get-id new-collection)))))
 
+(defn- fix-click-through
+  "Updates the dashboard card by pulling and renaming column behavior from the card."
+  [{id :id card :card_visualization dashcard :dashcard_visualization}]
+  (let [merged                   (merge dashcard card)
+        top-level-click-behavior (when (contains? merged "click")
+                                   {"type"         (get merged "click")
+                                    "linkType"     "url"
+                                    "linkTemplate" (get merged "click_link_template")})
+        column-settings          (get merged "column_settings")
+        updated-columns          (reduce-kv (fn [m col field-settings]
+                                              (if (and (contains? field-settings "view_as")
+                                                       (contains? field-settings "link_template"))
+                                                (assoc m col
+                                                       {"type"             (get field-settings "view_as")
+                                                        "linkType"         "url"
+                                                        "linkTemplate"     (get field-settings "link_template")
+                                                        "linkTemplateText" (get field-settings "link_text")} )
+                                                m))
+                                            {}
+                                            column-settings)]
+    ;; don't return anything if we don't have new stuff
+    (when (or top-level-click-behavior (seq updated-columns))
+      {:id id
+       :visualization_settings
+       (merge dashcard
+              (when top-level-click-behavior
+                {"click_behavior" top-level-click-behavior})
+              (when (seq updated-columns)
+                {"column_settings" updated-columns}))})))
+
+(defn- parse-to-json [& ks]
+  (fn [x]
+    (reduce #(update %1 %2 json/parse-string)
+            x
+            ks)))
+
+(defmigration ^{:author "dpsutton" :added "0.38.1"} migrate-click-through
```

Comment:
It would be helpful to have a comment explaining what's going on above the migration... see the one above it. Right now we all know exactly what it's doing and why, but when new people join the team it might not be so obvious what the purpose of this migration is or even what it's doing. It's better to be able to read a written description then to have to read all the code to try to understand what's going on